### DISCUSSION TITLE
FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory

### DISCUSSION THREAD

**Discussion by windowshopr** [open-webui/open-webui#4119] [General]
# Bug Report

## Description
**Bug Summary:**
Latest git repository as of July, 2024, running `docker compose up --pull always --build` in PowerShell of Windows 10, running Docker Desktop, results in the following traceback:

```
115.4 ✓ 2861 modules transformed.
121.5 rendering chunks...
143.4
143.4 <--- Last few GCs --->
143.4
143.4 [16:0x7fe77847d030]   104087 ms: Mark-Compact (reduce) 1948.6 (1994.4) -> 1948.6 (1994.9) MB, 1332.58 / 0.00 ms  (+ 45.0 ms in 0 steps since start of marking, biggest step 0.0 ms, walltime since start of marking 1404 ms) (average mu = 0.249, current mu = [16:0x7fe77847d030]   106348 ms: Mark-Compact (reduce) 1948.9 (1994.9) -> 1948.8 (1995.1) MB, 2017.23 / 0.00 ms  (average mu = 0.167, current mu = 0.108) allocation failure; GC in old space requested
143.4
143.4
143.4 <--- JS stacktrace --->
143.4
143.4 FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
143.4 ----- Native stack trace -----
143.4
144.4 npm ERR! path /app
144.4 npm ERR! command failed
144.4 npm ERR! signal SIGABRT
144.4 npm ERR! command sh -c npm run pyodide:fetch && vite build
144.4
144.4 npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2024-07-25T17_53_55_234Z-debug-0.log
------
failed to solve: process "/bin/sh -c npm run build" did not complete successfully: exit code: 1
```

**Steps to Reproduce:**
`git clone https://github.com/open-webui/open-webui.git`
`cd` into the directory
`docker compose up --pull always --build` with Docker Desktop running

**Expected Behavior:**
Container should build and run without error.

**Actual Behavior:**
[Describe what actually happened.]

## Environment

- **Open WebUI Version:** 0.3.10
- **Ollama (if applicable):** 0.3.0 Windows Preview

- **Operating System:** Windows 10
- **Browser (if applicable):** Chrome

I tried increasing the VMEMM limit from 2Gb to 4Gb in the .yaml file, however same issue.

**Comment by justinh-rahb** [comment]
Unless there's a specific reason, do you need to be building it? You could comment out the `build:` section of the compose file and drop the `--build` argument and it'll simply pull the images from GHCR.

**Reply by windowshopr** [reply]
Yep, in my case simply dropping the `--build` in the command worked. Thank you!

**Reply by m-primo** [reply]
> Unless there's a specific reason, do you need to be building it? You could comment out the `build:` section of the compose file and drop the `--build` argument and it'll simply pull the images from GHCR.

Because without the build command this happens:

> "Error response from daemon: Head "https://ghcr.io/v2/open-webui/open-webui/manifests/main": denied: denied"

This is the compose file for reference:

```yml
services:
    openwebui:
        image: ghcr.io/open-webui/open-webui:main
        ports:
            - 3000:8080
        extra_hosts:
            - host.docker.internal:host-gateway
        volumes:
            - ./data:/app/backend/data
```

**Reply by heblol** [reply]
Hi there, I also hit this problem, however, I do want to build it. What should I do to overcome this problem?

```
...
...
<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
----- Native stack trace -----
```

**Reply by alph4b3th** [reply]
In the Dockerfile, on line 34, there should be something like: "npm run build." Replace it with: "NODE_OPTIONS=--max-old-space-size=24576 npm run build"
